src/io/sqlite_store.rs

```diff
@@ -0,0 +1,193 @@
+use super::*;
+
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::Writeable;
+
+use rusqlite::{named_params, Connection};
+
+use std::fs;
+use std::io::Cursor;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+
+// The database file name.
+const SQLITE_DB_FILE: &str = "ldk_node.sqlite";
+
+// The table in which we store all data.
+const KV_TABLE_NAME: &str = "ldk_node_data";
+
+// The current SQLite `user_version`, which we can use if we'd ever need to do a schema migration.
+const SCHEMA_USER_VERSION: u16 = 1;
+
+/// A [`KVStore`] implementation that writes to and reads from an [SQLite] database.
+///
+/// [SQLite]: https://sqlite.org
+pub struct SqliteStore {
+	connection: Arc<Mutex<Connection>>,
+}
+
+impl SqliteStore {
+	pub(crate) fn new(dest_dir: PathBuf) -> Self {
+		fs::create_dir_all(dest_dir.clone()).unwrap_or_else(|_| {
+			panic!("Failed to create database destination directory: {}", dest_dir.display())
+		});
+		let mut db_file_path = dest_dir.clone();
+		db_file_path.push(SQLITE_DB_FILE);
+
+		let connection = Connection::open(db_file_path.clone()).unwrap_or_else(|_| {
+			panic!("Failed to open/create database file: {}", db_file_path.display())
+		});
+
+		connection
+			.pragma(Some(rusqlite::DatabaseName::Main), "user_version", SCHEMA_USER_VERSION, |_| {
+				Ok(())
+			})
+			.unwrap_or_else(|_| panic!("Failed to set PRAGMA user_version"));
+
+		let sql = format!(
+			"CREATE TABLE IF NOT EXISTS {} (
+			namespace TEXT NOT NULL,
```

Comment:
i think we should add the non-empty constraint on namespace as well, it is unexpected to have empty namespace, specially when we allow for clients to use custom schema with pluggable backend kvStore.

And instead use a static namespace in place of an empty one where needed.

Reply:
why do we want to have this inconsistency/workaround instead of opting for a simpler thing ?